### PR TITLE
Add useful comment on ServerInterface::Shutdown

### LINKED ISSUE #10324: Improve C++ documentation for Server::Shutdown
It's hard for users to understand the implications of Server::Shutdown with a deadline

### PR DESCRIPTION
**Pull request by vjpai**
Fixes #10324

### CODE REVIEW

File: include/grpcpp/impl/codegen/server_interface.h [hunk 1]
```diff
@@ -49,12 +49,28 @@ class ServerInterface : public internal::CallHook {
  public:
   virtual ~ServerInterface() {}
 
-  /// Shutdown the server, blocking until all rpc processing finishes.
-  /// Forcefully terminate pending calls after \a deadline expires.
+  /// \a Shutdown does the following things:
```
Comment (ncteisen) [inline include/grpcpp/impl/codegen/server_interface.h]: newline after this

File: include/grpcpp/impl/codegen/server_interface.h [hunk 2]
```diff
@@ -49,12 +49,28 @@ class ServerInterface : public internal::CallHook {
...
+  /// 1. Shutdown the server: deactivate all listening ports, mark it in
+  ///    "shutdown mode" so that further call Request's or incoming RPC matches
+  ///    are no longer allowed. Also return all Request'ed-but-not-yet-active
+  ///    calls as failed (!ok): note that this would even include default calls
```
Comment (ncteisen) [inline include/grpcpp/impl/codegen/server_interface.h]: I don't understand fully the last sentence. What types of calls are those?

File: include/grpcpp/impl/codegen/server_interface.h [hunk 3]
```diff
@@ -49,12 +49,28 @@ class ServerInterface : public internal::CallHook {
...
+  ///    added automatically by the C++ API without the user's input.
+  ///
+  /// 2. Block until all rpc method handlers invoked automatically by the sync
+  ///    API finish.
+  ///
+  /// 3. If all pending calls complete (and all their operations are
+  ///    retrieved by Next) before \a deadline expires, this finishes
+  ///    gracefully. Otherwise, forcefully cancel all pending calls associated
+  ///    with the server after \a deadline expires. In the case of the sync API,
+  ///    if the RPC function for a streaming call has already been started an
```
Comment (ncteisen) [inline include/grpcpp/impl/codegen/server_interface.h]: s/an/and

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]      +8  +0.0%

  [ = ]       0 TOTAL       +8  +0.0%



```

File: include/grpcpp/impl/codegen/server_interface.h [hunk 4]
```diff
@@ -49,12 +49,28 @@ class ServerInterface : public internal::CallHook {
...
+  ///    takes a week to complete, the RPC function won't be forcefully
+  ///    terminated (since that would leave state corrupt and incomplete).
```
Comment (ncteisen) [inline include/grpcpp/impl/codegen/server_interface.h]: So what would happen? Would the server not shutdown?

File: include/grpcpp/impl/codegen/server_interface.h (hunk 1 above)
Comment (vjpai) [reply include/grpcpp/impl/codegen/server_interface.h]: Done!

File: include/grpcpp/impl/codegen/server_interface.h (hunk 2 above)
Comment (vjpai) [reply include/grpcpp/impl/codegen/server_interface.h]: I've expanded the explanation

File: include/grpcpp/impl/codegen/server_interface.h (hunk 3 above)
Comment (vjpai) [reply include/grpcpp/impl/codegen/server_interface.h]: Done!

File: include/grpcpp/impl/codegen/server_interface.h (hunk 4 above)
Comment (vjpai) [reply include/grpcpp/impl/codegen/server_interface.h]: I've expanded the explanation.

File: include/grpcpp/impl/codegen/server_interface.h (hunk 1 above)
Comment (vjpai) [reply include/grpcpp/impl/codegen/server_interface.h]: Done!

**Comment by vjpai** [comment]
I've rebased and addressed reviewer comments. PTAL SVP.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Pull request merged**